mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/dto/DeletedMessageWithStorageInformationConverter.java

```diff
@@ -0,0 +1,94 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.vault.dto;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.james.blob.api.BlobId;
+import org.apache.james.blob.api.BucketName;
+import org.apache.james.core.MailAddress;
+import org.apache.james.core.MaybeSender;
+import org.apache.james.core.User;
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.vault.DeletedMessage;
+import org.apache.james.vault.metadata.DeletedMessageWithStorageInformation;
+import org.apache.james.vault.metadata.StorageInformation;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+import com.google.common.collect.ImmutableList;
+
+public class DeletedMessageWithStorageInformationConverter {
+    private final BlobId.Factory blobFactory;
+    private final MessageId.Factory messageIdFactory;
+    private final MailboxId.Factory mailboxIdFactory;
+
+    @Inject
+    public DeletedMessageWithStorageInformationConverter(BlobId.Factory blobFactory,
+                                                         MessageId.Factory messageIdFactory,
+                                                         MailboxId.Factory mailboxIdFactory) {
+        this.blobFactory = blobFactory;
+        this.messageIdFactory = messageIdFactory;
+        this.mailboxIdFactory = mailboxIdFactory;
+    }
+
+    public StorageInformation toStorageInformation(DeletedMessageWithStorageInformationDTO.StorageInformationDTO storageInformationDTO) {
```

Comment:
both input and output are actually `StorageInformation`
I would call methods `toDomainObject` and `toDTO` and use method overloading for handling different types